See https://gerrit.gromacs.org/376

From 3d2c02960f9f791fc7d95549b686c8b23651613b Mon Sep 17 00:00:00 2001
From: Christoph Junghans <junghans@votca.org>
Date: Mon, 19 Dec 2011 11:23:25 -0700
Subject: [PATCH] Set GMX_*_ASM true if GMX_ACCELERATION != auto

In case GMX_ACCELERATION was set to SSE manually (like in Gentoo Linux)
GMX_*_ASM would never be set (false), so sse kernel will not be build.

Change-Id: Icf04707c6174a0182fe4472536b568e1d2d91166
---
 CMakeLists.txt |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 25f22b7..b56aad1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -542,12 +542,6 @@ if (${GMX_ACCELERATION} STREQUAL "auto" AND NOT GMX_OPENMM)
 
     set(GMX_ACCELERATION "SSE" CACHE STRING "Accelerated kernels. Pick one of: auto, none, SSE, BlueGene, Power6, ia64, altivec, fortran" FORCE)
     
-    if (GMX_64_BIT)
-      set(GMX_X86_64_ASM ON CACHE BOOL "Add SSE assembly files for x86_64" FORCE)
-    else (GMX_64_BIT)
-      set(GMX_IA32_ASM ON CACHE BOOL "Add SSE assembly files for i386" FORCE)
-    endif (GMX_64_BIT)
-
     # Determine the assembler/compiler to use
   else()
     set(GMX_ACCELERATION "none" CACHE STRING "Accelerated kernels. Pick one of: auto, none, SSE, BlueGene, Power6, ia64, altivec, fortran" FORCE)
@@ -566,6 +560,12 @@ string(TOUPPER ${GMX_ACCELERATION} ${GMX_ACCELERATION})
 if(${GMX_ACCELERATION} STREQUAL "NONE")
   # nothing to do
 elseif(${GMX_ACCELERATION} STREQUAL "SSE")
+    
+    if (GMX_64_BIT)
+      set(GMX_X86_64_ASM ON CACHE BOOL "Add SSE assembly files for x86_64" FORCE)
+    else (GMX_64_BIT)
+      set(GMX_IA32_ASM ON CACHE BOOL "Add SSE assembly files for i386" FORCE)
+    endif (GMX_64_BIT)
 
     if(CMAKE_GENERATOR MATCHES "Visual Studio")
       option(GMX_ASM_USEASM-NASM "Use Nasm for assembly instead of compiler (necessary on windows)" ON)
-- 
1.7.8

